src/test/calculate.test.js

```diff
@@ -0,0 +1,174 @@
+import calculate from '../logic/caculate';
+
+it('All Clear', () => {
+  const obj = {
+    total: 12,
+    next: 12,
+    operation: 'x',
+  };
+  const buttonName = 'AC';
+  const result = calculate(obj, buttonName);
+  expect(result).toStrictEqual({
+    total: null,
+    next: null,
+    operation: null,
+  });
+});
+
+it('Is zero number', () => {
+  const obj = {
+    total: null,
+    next: 0,
+    operation: null,
+  };
+  const buttonName = '0';
+  const result = calculate(obj, buttonName);
+  expect(result).toStrictEqual({
+    total: null,
+    next: '0',
+  });
+});
+
+it('Is number and there is an operation', () => {
+  const obj = {
+    total: null,
+    next: 3,
```

Comment:
- [ ] It would be better professionally if you include all test cases that are related to the same topic or functionality under a `describe` block.
[Hint]: See that [video](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjpqMnXorL4AhWmiv0HHabkA3wQFnoECAQQAw&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DkVzw_f7TfCE&usg=AOvVaw0RUJ139efStEZZVJJ50P1z&cshid=1655392824985935) for more info about that.

Reply:
Thank you @Omar-Muhamad I have already added a description.